linera-client/src/client_context.rs

```diff
@@ -81,45 +84,54 @@ where
 
 #[cfg_attr(not(web), async_trait)]
 #[cfg_attr(web, async_trait(?Send))]
-impl<S, W> chain_listener::ClientContext for ClientContext<S, W>
+impl<S, W, K> chain_listener::ClientContext for ClientContext<S, W, K>
```

Comment:
This PR becomes _a lot_ smaller if we use a trait object here rather than making it part of the type.

Reply:
Updated.